ambry-store/src/main/java/com.github.ambry.store/DiskManager.java

```diff
@@ -61,16 +63,17 @@
    */
   DiskManager(DiskId disk, List<ReplicaId> replicas, StoreConfig config, ScheduledExecutorService scheduler,
       StorageManagerMetrics metrics, StoreKeyFactory keyFactory, MessageStoreRecovery recovery,
-      MessageStoreHardDelete hardDelete, Time time) {
+      MessageStoreHardDelete hardDelete, Time time) throws StoreException {
     this.disk = disk;
     this.metrics = metrics;
     this.time = time;
     diskIOScheduler = new DiskIOScheduler(getThrottlers(config, time));
+    diskSpaceAllocator = new DiskSpaceAllocator(new File(disk.getMountPath(), config.storeReserveFileDirName));
```

Comment:
added checking in `StoreManager`, since this is where sanity checking for `StoreConfig` is done.